src/Dapr.Actors/Runtime/ReminderInfo.cs

```diff
@@ -36,82 +36,62 @@ internal class ReminderInfo
 
         public byte[] Data { get; private set; }
 
-        internal static ReminderInfo Deserialize(Stream stream)
+        internal static async Task<ReminderInfo> DeserializeAsync(Stream stream)
         {
-            // Deserialize using JsonReader as we know the property names in response. Deserializing using JsonReader is most performant.
+            var serializeOptions = new JsonSerializerOptions();
             using var streamReader = new StreamReader(stream);
-            using var reader = new JsonTextReader(streamReader);
-            return reader.Deserialize(GetFromJsonProperties);
-        }
+            var text = await streamReader.ReadToEndAsync();
```

Comment:
if It’s always utf8 (which you’re defaulting to right now) you can just pass the stream directly to the serializer. You can fallback to the string for non-utf8 encodings